### PR TITLE
Add repository contextual menu items

### LINKED ISSUE #1554: Contextual Menu (right click) to open a repo in Finder, Atom, or external editor.
### Description

I would like to open a repo in another application.

### Version

**GitHub Desktop version:** 0.5.3

**OS version:** macOS 10.12.4 (16E195)

### Steps to Reproduce

1. Click the current repo drop down.
2. Right click on a repo

**Expected behaviour:**
A contextual menu shows with options to open a repo in Finder, Atom, or external editor (maybe this last part would be supported via a plugin?).

**Actual behaviour:**
The only option shown is to remove the repo.

Github for Mac currently implements this functionality.

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #1554

### CODE REVIEW

**Comment by joshaber** [comment]
🏷 ⏩

File: app/src/main-process/menu/build-default-menu.ts [hunk 1]
```diff
@@ -211,7 +211,7 @@ export function buildDefaultMenu(sharedProcess: SharedProcess): Electron.Menu {
         click: emit('open-in-shell'),
       },
       {
-        label: __DARWIN__ ? 'Open in Finder' : '&Open in Explorer',
+        label: __DARWIN__ ? 'Show in Finder' : '&Show in Explorer',
```
Comment (shimpossible) [inline app/src/main-process/menu/build-default-menu.ts]: This conflicts with line 221
''' label: __DARWIN__ ? 'Repository Settings…' : 'Repository &settings…', '''
Can't have both map to 'S' in windows.  Maybe  Open in &Explorer

File: app/src/ui/repositories-list/repository-list-item.tsx [hunk 2]
```diff
@@ -1,12 +1,20 @@
 import * as React from 'react'
 import { Repository } from '../../models/repository'
 import { Octicon, iconForRepository } from '../octicons'
-import { CloningRepository } from '../../lib/dispatcher'
-import { showContextualMenu } from '../main-process-proxy'
+import { showContextualMenu, IMenuItem } from '../main-process-proxy'
+import { Repositoryish } from './group-repositories'
```
Comment (iAmWillShepherd) [inline app/src/ui/repositories-list/repository-list-item.tsx]: I know this is not really a part of this PR, but can we come up with a better name for `Repositoryish`? IMO, it makes more sense to call rename `Repository` -> `LocalRepository` and `Repositoryish` -> `Repository`.

File: app/src/main-process/menu/build-default-menu.ts (hunk 1 above)
Comment (joshaber) [inline app/src/main-process/menu/build-default-menu.ts]: Great catch!

File: app/src/ui/repositories-list/repository-list-item.tsx (hunk 2 above)
Comment (joshaber) [inline app/src/ui/repositories-list/repository-list-item.tsx]: Yeah I'm into that 👍 Mind creating a `tech-debt` 🏷'd issue for that?

**Comment by joshaber** [comment]
🤵

File: app/src/main-process/menu/build-default-menu.ts (hunk 1 above)
Comment (shimpossible) [inline app/src/main-process/menu/build-default-menu.ts]: &E doesn't work by the way.. it conflicts with something else.

File: app/src/main-process/menu/build-default-menu.ts [hunk 3]
```diff
@@ -211,7 +211,7 @@ export function buildDefaultMenu(sharedProcess: SharedProcess): Electron.Menu {
         click: emit('open-in-shell'),
       },
       {
-        label: __DARWIN__ ? 'Show in Finder' : '&Show in Explorer',
+        label: __DARWIN__ ? 'Show in Finder' : 'Show in &Explorer',
```
Comment (shimpossible) [inline app/src/main-process/menu/build-default-menu.ts]: Sorry to mislead you earlier.  I hadn't tested my suggesting.  &E also conflicts.  
But &X works 👍

Comment (joshaber) [inline app/src/main-process/menu/build-default-menu.ts]: That's what I get for being hasty 😆

**Comment by joshaber** [comment]
🥖

File: app/src/ui/repositories-list/repository-list-item.tsx (hunk 2 above)
Comment (iAmWillShepherd) [inline app/src/ui/repositories-list/repository-list-item.tsx]: Sure thing, see issue #2031

**Pull request merged**